Mathlib/RingTheory/MvPolynomial/WeightedHomogeneous.lean

```diff
@@ -409,20 +418,45 @@ theorem sum_weightedHomogeneousComponent :
   rw [finsum_eq_sum _ (weightedHomogeneousComponent_finsupp φ)]
   ext1 d
   simp only [coeff_sum, coeff_weightedHomogeneousComponent]
-  rw [Finset.sum_eq_single (weightedDegree' w d)]
+  rw [Finset.sum_eq_single (weightedDegree w d)]
   · rw [if_pos rfl]
   · intro m _ hm'
     rw [if_neg hm'.symm]
   · intro hm
     rw [if_pos rfl]
     simp only [Finite.mem_toFinset, mem_support, Ne.def, Classical.not_not] at hm
-    have := coeff_weightedHomogeneousComponent (w := w) (weightedDegree' w d) φ d
+    have := coeff_weightedHomogeneousComponent (w := w) (weightedDegree w d) φ d
     rw [hm, if_pos rfl, coeff_zero] at this
     exact this.symm
 #align mv_polynomial.sum_weighted_homogeneous_component MvPolynomial.sum_weightedHomogeneousComponent
 
+theorem finsum_weightedHomogeneousComponent :
+    (finsum fun m => weightedHomogeneousComponent w m φ) = φ := by
+  rw [sum_weightedHomogeneousComponent]
+
 variable {w}
 
+theorem IsWeightedHomogeneous.weightedHomogeneousComponent_same
+    [DecidableEq M] (m : M) (p : MvPolynomial σ R) (hp : IsWeightedHomogeneous w p m) :
```

Comment:
Do you need `DecidableEq M` in the statement?

Reply:
Good catch, I don't. Thanks!